Add mechanism to display textual holdings (866 MARC field)

alexklbuckley opened the pull request:
Sponsored-By: Aucklan University of Technology, New Zealand

alexklbuckley committed:
Display textual holdings (866 MARC field) in core record data

Sponsored-By: Auckland University of Technology, New Zealand

alexklbuckley committed:
Fixing unit tests

Sponsored-By: Auckland University of Technology, New Zealand

demiankatz commented on module/VuFind/tests/unit-tests/src/VuFindTest/View/Helper/Root/RecordDataFormatterTest.php:
```diff
@@ -521,9 +521,9 @@ public function testFormatting(string $function): void
-            'Series' => 'Vico, Giambattista, 1668-1744. Works. 1982 ;',
             'Multi Count' => 1,
             'Multi Data' => 'Book',
+            'Series' => 'Vico, Giambattista, 1668-1744. Works. 1982 ;',
```
Is this change related to the rest of the PR? It certainly doesn't hurt to put the expectations in alphabetical order, but I don't see a reason why your other code changes would require this test to be updated. Did you start to work on adding a test case for the new functionality but decide not to proceed with it?

alexklbuckley replied:
thanks @demiankatz yes this was due to me starting to work on adding a test case for the new functionality and then I decided not to proceed with it. I've removed this change in my latest follow up

demiankatz commented on module/VuFind/src/VuFind/View/Helper/Root/RecordDataFormatterFactory.php:
```diff
@@ -333,6 +333,10 @@ public function getDefaultCoreSpecs()
                 'itemSuffix' => '</span>',
             ]
         );
+        $spec->setLine(
```
I wonder if it would be safer to make this an example in RecordDataFormatter.ini instead of turning it on by default. Since 866 is technically part of MARC holdings, it's not really a standards-compliant thing to do by default. I realize this is a common use case because of merging holdings records into bib records, but I imagine turning it on automatically might have unwanted/unexpected side effects for many people. Making it easy to activate while leaving it off to begin with seems like the safest path forward (though I'm open to discussing further if you or others feel differently).

alexklbuckley replied:
thanks @demiankatz . Yes, that makes perfect sense. In my latest follow-up I've shifted this over to be an example in the RecordDataFormatter.ini file. As I am not very familiar with that file I am not sure if I have added it correctly. If you wouldn;t mind checking?

demiankatz reviewed (changes requested):
See below for a couple of initial thoughts...

alexklbuckley committed:
Merge branch 'vufind-org:dev' into textualholdings

alexklbuckley committed:
Merge branch 'vufind-org:dev' into textualholdings

alexklbuckley committed:
Display textual holdings (866 MARC field)

- Remove redundant unit test change.
- Switch to adding the textual holdings (866) as an example in the
  RecordDataFormatter.ini file, rather than having it automatically on.

Sponsored-by: Auckland University of Technology, New Zealand

demiankatz committed:
Minor adjustments to example.

demiankatz committed:
Fix comment to match.

demiankatz reviewed (approved):
Thanks, @alexklbuckley! Your example works for me, though I made a couple of minor adjustments -- changing "Holdings" to "TextualHoldings" to be more specific, and adding `pos = 2000` so the new field displays at the end of the record instead of the beginning.

alexklbuckley commented:
thanks so much @demiankatz ! Appreciate all your help as always!
